src/ast/ast_common.hpp

```diff
@@ -1496,6 +1521,40 @@ struct Ast: public std::enable_shared_from_this<Ast> {
     virtual bool is_ontology_statement() {
         return false;
     }
+
+    /**
+     *\brief Parent getter
+     *
+     * returning a raw pointer may create less problems that the
+     * shared_from_this from the parent.
+     *
+     * Check Ast::parent for more information
+     */
+    inline virtual Ast* get_parent() const {
+        return parent;
+    }
+
+    /**
+     *\brief Parent setter
+     *
+     * Usually, the parent parent pointer cannot be set in the constructor
+     * because children are generally build BEFORE the parent. Conversely,
+     * we set children parents directly in the parent constructor using
+     * set_parent_in_children()
+     *
+     * Check Ast::parent for more information
+     */
+    inline virtual void set_parent(Ast* p) {
```

Comment:
inline not required (member functions in class definitions are inline by default).

Reply:
thx

Reply:
@cattabiani : you marked resolved but it seems like change has not been applied? i.e. `inline` should be removed. I will push a commit with the changes.